query-planner-js/src/buildPlan.ts

```diff
@@ -2492,6 +2497,25 @@ export class QueryPlanner {
     };
     this._lastGeneratedPlanStatistics = statistics;
 
+    if (this.config.debug.bypassPlannerForSingleSubgraph) {
+      // A federated query graph always have 1 more sources than there is subgraph, because the root vertices
+      // belong to no subgraphs and use a special source named '_'. So we skip that "fake" source.
+      const subgraphs = mapKeys(this.federatedQueryGraph.sources).filter((name) => name !== FEDERATED_GRAPH_ROOT_SOURCE);
+      if (subgraphs.length === 1) {
```

Comment:
Do you want the else case to error?

Reply:
No. I think the config name and documentation makes it clear that it only applies to single subgraphs, and it's clearly flagged as for debug/testing so I'm not really worried about someone not understanding why they don't get what they want out of this. On the flip side, erroring could make it more annoying to use as you'd have to remember to disable the option when you go from 1 to 2 subgraph, or get a runtime crash. Admittedly this is all a bit subjective since again, it is clearly flagged for debug/testing, but I personally markedly prefer it not having a `else`.